### PR TITLE
[SPARK-35583][DOCS] Move JDBC data source options from Python and Scala into a single page

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes move missing JDBC data source options from Python, Scala and Java into a single page.

### Why are the changes needed?

So far, the documentation for JDBC data source options is separated into different pages for each language API documents. However, this makes managing many options inconvenient, so it is efficient to manage all options in a single page and provide a link to that page in the API of each language.

### Does this PR introduce _any_ user-facing change?

Yes, the documents will be shown below after this change:

- "JDBC To Other Databases" page
Screen Shot 2021-06-02 at 11 34 14 AM

- Python
Screen Shot 2021-06-01 at 2 57 40 PM

- Scala
Screen Shot 2021-06-01 at 2 57 03 PM

- Java
Screen Shot 2021-06-01 at 2 58 27 PM

### How was this patch tested?

Manually build docs and confirm the page.

### CODE REVIEW

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -627,8 +627,6 @@ def jdbc(self, url, table, column=None, lowerBound=None, upperBound=None, numPar
 
         Parameters
         ----------
-        url : str
-            a JDBC URL of the form ``jdbc:subprotocol:subname``
         table : str
             the name of the table
         column : str, optional
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:632]: I think we can remove `lowerBound`, `upperBound`, and `numPartitions`.
And, fix the description of `column` to something like:

Alias of `partitionColumn` option. Refer to `partitionColumn` in `Data Source Option <...>`_ in the version you use.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 2]
```diff
@@ -301,23 +306,22 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
...
    * @param table Name of the table in the external database.
-   * @param columnName the name of a column of numeric, date, or timestamp type
-   *                   that will be used for partitioning.
-   * @param lowerBound the minimum value of `columnName` used to decide partition stride.
-   * @param upperBound the maximum value of `columnName` used to decide partition stride.
-   * @param numPartitions the number of partitions. This, along with `lowerBound` (inclusive),
-   *                      `upperBound` (exclusive), form partition strides for generated WHERE
-   *                      clause expressions used to split the column `columnName` evenly. When
-   *                      the input is less than 1, the number is set to 1.
+   * @param columnName alias of `partitionColumn` option. Refer to `partitionColumn` in
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:314]:
```suggestion
   * @param columnName Alias of `partitionColumn` option. Refer to `partitionColumn` in
```

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 3]
```diff
@@ -754,6 +746,8 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    *                             or "SERIALIZABLE", corresponding to standard transaction
    *                             isolation levels defined by JDBC's Connection object, with default
    *                             of "READ_UNCOMMITTED".
+   *
+   *
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:750]: Let's remove these empty lines

**Comment by HyukjinKwon** [comment]
cc @maropu FYI

**Comment by maropu** [comment]
The proposal looks fine. Have you checked `R` ones? e.g., https://github.com/apache/spark/blob/master/R/pkg/R/SQLContext.R#L612-L621

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 4]
```diff
@@ -301,23 +306,22 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * Don't create too many partitions in parallel on a large cluster; otherwise Spark might crash
    * your external database systems.
    *
-   * @param url JDBC database url of the form `jdbc:subprotocol:subname`.
+   * You can find the JDBC-specific options for reading table via JDBC in
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:309]: Can we change: "JDBC-specific options for reading table" -> "JDBC-specific option and parameter documentation for reading tables"?

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 5]
```diff
@@ -282,6 +282,10 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * Construct a `DataFrame` representing the database table accessible via JDBC URL
    * url named table and connection properties.
    *
+   * You can find the JDBC-specific options for reading table via JDBC in
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:285]: reading a table or reading tables

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Comment by itholic** [comment]
Thanks, @HyukjinKwon .

Also updated the PR description!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by maropu** [comment]
Thank you for this update, @itholic late lgtm.

**Pull request closed without merging**